Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -568,6 +568,7 @@ public class PBXProjGenerator {
             (try path.children().sorted(), path)
 
         let excludedFiles: [String] = [".DS_Store"]
+        let sourceExcludedFiles: [String] = source.excludes.map { "\(source.path)/\($0)" }
```

Comment:
This is a good start. Though ideally I think we'd like to be able to specify excludes via a glob pattern (eg `**.DS_Store`) PathKit has build in support for glob patterns. 
Or at the very least to specify certain file types which could be done with some simple custom logic `*.filetype`

And would you want to exclude a folder? I don't know, maybe it's not useful?